cli/flox/src/commands/services/stop.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::Result;
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::providers::services::stop_services;
+use tracing::instrument;
+
+use crate::commands::{environment_select, EnvironmentSelect};
+use crate::subcommand_metric;
+use crate::utils::message;
+
+#[derive(Bpaf, Debug, Clone)]
+pub struct Stop {
+    #[bpaf(external(environment_select), fallback(Default::default()))]
+    environment: EnvironmentSelect,
+
+    /// Names of the services to stop
+    #[bpaf(positional("name"))]
+    names: Vec<String>,
+}
+
+impl Stop {
+    // TODO: are these nested services->stop?
+    #[instrument(name = "stop", skip_all)]
+    pub async fn handle(self, flox: Flox) -> Result<()> {
+        // TODO: include spaces?
+        subcommand_metric!("services stop");
+
+        let env = self
+            .environment
+            .detect_concrete_environment(&flox, "Services in")?
+            .into_dyn_environment();
+        let socket = env.services_socket_path(&flox)?;
+
+        stop_services(socket, &self.names)?;
+
+        message::updated("Stop! In the name of love");
```

Comment:
This was a placeholder. Should we output the names of the services that we stopped?

We should add assertions to the happy-path tests once we decide on the messaging.

cc @jennymahmoudi

![giphy](https://github.com/flox/flox/assets/260438/8220bfbc-3af5-4e33-932c-aac132a8c87b)

Reply:
From sync with Zach:
- output the names of the services that were stopped for the happy paths
- later followup, we can:
  - query `ProcessStates` afterwards to see if any services were stopped when there are also errors
  - query `ProcessStates` beforehand to distinguish missing from stopped services

Reply:
https://github.com/flox/flox/issues/1771